src/utils/filter-utils.js

```diff
@@ -840,3 +969,51 @@ export function getDatasetFieldIndexForFilter(dataset, filter) {
   return notNullorUndefined(fieldIndex) ? fieldIndex : -1;
 }
 
+/**
+ * Generates polygon filter
+ * @param layers array of layers
+ * @param feature polygon to use
+ * @return {object} filter
+ */
+export function generatePolygonFilter(layers, feature) {
+  const {dataId, layerId, name} = layers.reduce((acc, layer) => ({
+    ...acc,
+    dataId: [...acc.dataId, layer.config.dataId],
+    layerId: [...acc.layerId, layer.id],
+    name: [...acc.name, layer.config.label]
+  }), {
+    dataId: [],
+    layerId: [],
+    name: []
+  });
+
+  return updatePolygonFilter({
+    ...getDefaultFilter(dataId),
+    fixedDomain: true,
+    type: FILTER_TYPES.polygon,
+    name,
+    layerId
+  }, feature);
+}
+
+/**
+ * Updates polygon filter
+ * @param filter
+ * @param feature
+ * @return {object} Filter
+ */
+export function updatePolygonFilter(filter, feature) {
+  const polygon = turfPolygon(feature.geometry.coordinates);
```

Comment:
a turfPolygon is a geojson Polygon feature, same as the `feature` object react-map-gl-draw returns. Here you dn't need to call  `turfPolygon` and merge them. this works:
```js
export function updatePolygonFilter(filter, feature) {
  // const polygon = turfPolygon(feature.geometry.coordinates);
  return {
    ...filter,
    // we merge both turf and feature properties into one
    value: {
      id: feature.id,
      ...feature
    }
  }
}
```